acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/Eth2CustomNetworkFileAcceptanceTest.java

```diff
@@ -0,0 +1,88 @@
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.tests.signing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import tech.pegasys.teku.bls.BLS;
+import tech.pegasys.teku.bls.BLSKeyPair;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.bls.BLSSecretKey;
+import tech.pegasys.teku.bls.BLSSignature;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.teku.spec.datastructures.util.ForkAndSpecMilestone;
+import tech.pegasys.web3signer.core.service.http.handlers.signing.eth2.Eth2SigningRequestBody;
+import tech.pegasys.web3signer.dsl.utils.Eth2BlockSigningRequestUtil;
+import tech.pegasys.web3signer.dsl.utils.MetadataFileHelpers;
+import tech.pegasys.web3signer.signing.KeyType;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.io.Resources;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class Eth2CustomNetworkFileAcceptanceTest extends SigningAcceptanceTestBase {
+  private static final Path NETWORK_CONFIG_PATH =
+      Path.of(Resources.getResource("eth2/network_config.yaml").getPath());
+  private static final String PRIVATE_KEY =
+      "3ee2224386c82ffea477e2adf28a2929f5c349165a4196158c7f3a2ecca40f35";
+
+  private static final MetadataFileHelpers metadataFileHelpers = new MetadataFileHelpers();
+  private static final BLSSecretKey key =
+      BLSSecretKey.fromBytes(Bytes32.fromHexString(PRIVATE_KEY));
+  private static final BLSKeyPair keyPair = new BLSKeyPair(key);
+  private static final BLSPublicKey publicKey = keyPair.getPublicKey();
+
+  @BeforeEach
+  void setup() {
+    final String configFilename = publicKey.toString().substring(2);
+    final Path keyConfigFile = testDirectory.resolve(configFilename + ".yaml");
+    metadataFileHelpers.createUnencryptedYamlFileAt(keyConfigFile, PRIVATE_KEY, KeyType.BLS);
+  }
+
+  @Test
+  void signAndVerifyBlockV2SignatureForAllEnabledMilestones() throws Exception {
+    setupEth2SignerWithCustomNetworkConfig(NETWORK_CONFIG_PATH);
+
+    final Spec spec = SpecFactory.create(NETWORK_CONFIG_PATH.toString());
+    final List<ForkAndSpecMilestone> enabledMilestones = spec.getEnabledMilestones();
+    assertThat(enabledMilestones.size()).isEqualTo(3);
+
+    for (final ForkAndSpecMilestone enabledMilestone : enabledMilestones) {
+      final UInt64 forkEpoch = enabledMilestone.getFork().getEpoch();
+      final UInt64 startSlot = spec.computeStartSlotAtEpoch(forkEpoch);
+      assertResponse(spec, forkEpoch, startSlot);
+    }
```

Comment:
Nice, this seems cleaner 👍 . So I guess we don't care about all the slots, just that the first slot of the fork epoch returns the correct spec?